src/containers/stroke-color-indicator.jsx

```diff
@@ -15,9 +15,24 @@ class StrokeColorIndicator extends React.Component {
         bindAll(this, [
             'handleChangeStrokeColor'
         ]);
+
+        // Flag to track whether an svg-update-worthy change has been made
+        this._hasChanged = false;
+    }
+    componentWillReceiveProps (newProps) {
+        const {strokeColorModalVisible, onUpdateSvg} = this.props;
+        if (strokeColorModalVisible && !newProps.strokeColorModalVisible) {
+            // Submit the new SVG, which also stores a single undo/redo action.
+            if (this._hasChanged) onUpdateSvg();
+        } else if (newProps.strokeColorModalVisible && !strokeColorModalVisible) {
+            // If modal is opening, reset the _hasChanged flag to false;
+            this._hasChanged = false;
```

Comment:
You could put this in the if statement and skip the else if (in both places)

Reply:
Fixed in 3dab6ef